### PR TITLE
Add development flow document

### PR DESCRIPTION
**Pull request by mestery**
Signed-off-by: Kyle Mestery <mestery@mestery.com>

### CODE REVIEW

File: DEVELOPMENT_FLOW.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+This document attempts to document the development flow as the project moves
+towards code generation.
+
+## Sub-Projects and GitHub Repositories
+
+Each sub-project has their own GitHub Repository where they are currently
+iterating on documents and some early code. We advocate that model should
+continue going forward. Further, if a sub-project wants to create additional
+GitHub repositories, that should be fine as well.
```
Comment (glimchb) [inline DEVELOPMENT_FLOW.md:11]: any naming conventions if we want to create more repos for the same sub-group ?

File: DEVELOPMENT_FLOW.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
+# OPI Project Development Flow
+
+This document attempts to document the development flow as the project moves
+towards code generation.
+
+## Sub-Projects and GitHub Repositories
+
+Each sub-project has their own GitHub Repository where they are currently
```
Comment (glimchb) [inline DEVELOPMENT_FLOW.md:8]: we should say something about external public repos... guidelines... and maybe licensing...

File: DEVELOPMENT_FLOW.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+## Validation Across Repositories
+
+We expect each sub-project to have appropriate CI testing:
+
+* Linting of various forms (markdown, code specific linting)
+* Unit tests
+* Functional tests
+* Container builds and pushed to GitHub Container Registry
+
+We also expect testing across repositories to work such that sub-projects can
```
Comment (glimchb) [inline DEVELOPMENT_FLOW.md:31]: today there is a lot of copy-paste between repos... for example checkers and linters and more... we might need a script to update and push something to all repos from main, for example... wdyt ?

**Review by glimchb — COMMENTED** [review]
lgtm, comments inline

**Review by glimchb — APPROVED** [review]
lgtm, few small nits, can be addressed later

File: DEVELOPMENT_FLOW.md (hunk 1 above)
Comment (seroyer) [reply DEVELOPMENT_FLOW.md:11]: I don't think we need to necessarily tie the name of a repo to the subgroup that initially created it.  Each repo should be accurately named for its purpose.

File: DEVELOPMENT_FLOW.md (hunk 3 above)
Comment (seroyer) [reply DEVELOPMENT_FLOW.md:31]: Looks like it's possible to share workflows between repos: https://docs.github.com/en/actions/using-workflows/reusing-workflows
So we should be able to have one repo host those common things and they'd just be referenced from the others.

**Review by seroyer — APPROVED** [review]
Looks good!  Thanks for doing this!

File: DEVELOPMENT_FLOW.md (hunk 1 above)
Comment (glimchb) [reply DEVELOPMENT_FLOW.md:11]: I mean, not that I advocate for it, like mandatory `opi-` prefix on each repo name

**Pull request merged**